Quantum-Control-Applications/ESR/T2.py

```diff
@@ -0,0 +1,165 @@
+from qm import SimulationConfig
+from qm.qua import *
+from qm import LoopbackInterface
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from configuration import *
+import matplotlib.pyplot as plt
+from qm.simulate.credentials import create_credentials
+
+################################
+# Open quantum machine manager #
+################################
+
+qmm = QuantumMachinesManager(host=qop_ip, port=qop_port)
+
+########################
+# Open quantum machine #
+########################
+
+qm = qmm.open_qm(config)
+
+###################
+# The QUA program #
+###################
+
+pi_len = int(320//4)
+pi_half_len = int(160//4)
+
+delay_min = int(1500//4)
+delay_max = int(20000//4)
+ddelay = 1000//4
+
+delays = np.arange(delay_min, delay_max + ddelay/2, ddelay)
+
+cooldown_time = int(10e6//4)
+
+n_avg = 100
+
+with program() as T2:
+
+    n = declare(int)
+    n_st = declare_stream()
+    delay_len = declare(int)
+    pulse_delay = declare(int)
+    readout_delay = declare(int)
+
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+
+    with for_(n, 0, n < n_avg, n + 1):
+
+        with for_(delay_len, delay_min, delay_len <= delay_max, delay_len + ddelay):
+
+            # wait(cooldown_time)
+
+            reset_phase('ensemble')
+            reset_phase('resonator')
+            reset_frame('ensemble')
+
+            echo = declare_stream(adc_trace=True)
+
+            assign(pulse_delay, delay_len - Cast.mul_int_by_fixed(pi_half_len, 0.5) - Cast.mul_int_by_fixed(pi_len, 0.5))
+            assign(readout_delay, delay_len - Cast.mul_int_by_fixed(pi_len, 0.5) - Cast.mul_int_by_fixed(readout_len, 0.125) - 5)
+
```

Comment:
here delay_len changes.
But still, no need for Cast.mul_int_by_fixed

Reply:
Is it no need because it is done in python? so then it know to do well (int * float)?

Reply:
Nice! I checked with the simulation and the calculations comes out right thanks!